app/mutations/addresses/release.rb

```diff
@@ -31,10 +32,18 @@ def execute
         info "Skip gateway address=#{address.id} in pool=#{@pool.id}"
 
       else
-        info "Delete address=#{address.id} in pool=#{@pool.id}"
-
-        address.delete!
+        if ping?(address.address)
+          info "#{self.address} still responds to ping, deleting omitted"
```

Comment:
It wasn't released. So I think it should be some kind of "4xx temporary error try again later" kind of case.